vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -295,8 +297,8 @@ - (BOOL)hasViewedAdsInPreviousCycle
 - (void)reportLoadedPageWithURL:(NSURL *)url innerText:(NSString *)text
 {
   if (![self isAdsServiceRunning]) { return; }
-  const auto urlString = std::string(url.absoluteString.UTF8String);
-  ads->OnPageLoaded(urlString, std::string(text.UTF8String));
+  const auto urlString = base::SysNSStringToUTF8(url.absoluteString);
+  ads->OnPageLoaded(urlString, base::SysNSStringToUTF8(text));
```

Comment:
Should we always use `SysNSStringToUTF8` instead of `.UTF8String` or this is needed only for scope of this PR?

Reply:
I've added to the few points where I think we may have UTF16 strings come in, but it's probably overkill. Most of the strings we use will always be utf8, but in the few cases we take external strings that don't have restrictions  such as html or text from web pages I swapped them over.

Adding them to URLs was probably overkill to, not sure if its really needed there since I think you cannot use UTF16 in URLs

Reply:
thanks for the explanation

Reply:
I have one question: Is this conversion for the text only? IE: is it necessary for the URL too?
Reason I'm asking is because if it's text, it might also be chunked encoding and that's possibly the reason it failed..

But nice solution! :)